### PR TITLE
Added Support for downloading JDBC driver from an artifactory which is secured with authentication

### PR DESCRIPTION
**Pull request by Heshma**
Added Support for downloading JDBC driver from an artifactory which is secured with authentication

### CODE REVIEW

File: charts/pega/values-large.yaml [hunk 1]
```diff
@@ -48,6 +48,20 @@ global:
     dataSchema: "YOUR_DATA_SCHEMA"
     customerDataSchema: ""
 
+  customArtifactory:
```
Comment (taz-pega-work) [inline charts/pega/values-large.yaml:51]: customArtifactory:
    # If you use a secured custom artifactory to manager your JDBC driver, provide the authentication details below by filling in the appropriate authentication section, either basic or apiKey.
    authentication:
      # Provide the basic authentication credentials or the API key authentication details to satisfy your custom artifactory authentication mechanism.
      basic:
        username: "YOUR_ARTIFACTORY_USERNAME"
        password: "YOUR_ARTIFACTORY_PASSWORD"
      apiKey:
        headerName: "YOUR_ARTIFACTORY_DEDICATED_APIKEY_HEADER"
        value: "YOUR_ARTIFACTORY_API_KEY"
    # Leave `customArtifactory.enableSSLVerification` enabled to ensure secure access to your custom artifactory; when `customArtifactory.enableSSLVerification` is `false`, SSL verification is skipped and establishes an insecure connection.
enableSSLVerification: true
      # Provide a required domain certificate for your custom artifactory server; if none is required, leave this field blank.
    certificate:

File: charts/pega/values-minimal.yaml [hunk 2]
```diff
@@ -44,6 +44,20 @@ global:
     dataSchema: "YOUR_DATA_SCHEMA"
     customerDataSchema: ""
 
+  customArtifactory:
```
Comment (taz-pega-work) [inline charts/pega/values-minimal.yaml:47]: same comment as for other yaml>
customArtifactory:
If you use a secured custom artifactory to manager your JDBC driver, provide the authentication details below by filling in the appropriate authentication section, either basic or apiKey.

authentication:

Provide the basic authentication credentials or the API key authentication details to satisfy your custom artifactory authentication mechanism.

basic:
username: "YOUR_ARTIFACTORY_USERNAME"
password: "YOUR_ARTIFACTORY_PASSWORD"
apiKey:
headerName: "YOUR_ARTIFACTORY_DEDICATED_APIKEY_HEADER"
value: "YOUR_ARTIFACTORY_API_KEY"

Leave customArtifactory.enableSSLVerification enabled to ensure secure access to your custom artifactory; when customArtifactory.enableSSLVerification is false, SSL verification is skipped and establishes an insecure connection.

enableSSLVerification: true

Provide a required domain certificate for your custom artifactory server; if none is required, leave this field blank.
certificate:

File: charts/pega/values.yaml [hunk 3]
```diff
@@ -50,17 +50,17 @@ global:
     customerDataSchema: ""
 
   customArtifactory:
-    # If using a custom artifactory to download JDBC uri, supply the authentication details here.
-    auth:
+    # If using a custom artifactory secured with Basic or APIKey Authentication to download JDBC driver, supply the authentication details here.
```
Comment (taz-pega-work) [inline charts/pega/values.yaml:53]: If you use a secured custom artifactory to manager your JDBC driver, provide the authentication details below by filling in the appropriate authentication section, either basic or apiKey.

File: charts/pega/values.yaml [hunk 4]
```diff
@@ -50,17 +50,17 @@ global:
...
+    authentication:
       # Provide either basic authentication or api key authentication details based on preferred way of authentication mechanism to custom artifactory.
```
Comment (taz-pega-work) [inline charts/pega/values.yaml:55]: Provide the basic authentication credentials or the API key authentication details to satisfy your custom artifactory authentication mechanism.

File: charts/pega/values.yaml [hunk 5]
```diff
@@ -50,17 +50,17 @@ global:
...
       basic:
         username: "YOUR_ARTIFACTORY_USERNAME"
         password: "YOUR_ARTIFACTORY_PASSWORD"
       apiKey:
         headerName: "YOUR_ARTIFACTORY_DEDICATED_APIKEY_HEADER"
         value: "YOUR_ARTIFACTORY_API_KEY"
-    enable_ssl_verification: true
-    #Custom artifactory domain server certificate if needed.
+    enableSSLVerification: true
```
Comment (taz-pega-work) [inline charts/pega/values.yaml:62]: Leave customArtifactory.enableSSLVerification enabled to ensure secure access to your custom artifactory; when customArtifactory.enableSSLVerification is false, SSL verification is skipped and establishes an insecure connection.

enableSSLVerification: true

Provide a required domain certificate for your custom artifactory server; if none is required, leave this field blank.
certificate:

File: docs/Deploying-Pega-on-AKS.md [hunk 6]
```diff
@@ -482,6 +482,8 @@ helm inspect values pega/pega > <local filepath>/aks-demo/pega.yaml
...
 | jdbc.rulesSchema: jdbc.dataSchema:  | Set the names of both your rules and the data schema to the values that Pega Platform uses for these two schemas.      | rulesSchema: "rules" dataSchema: "data" |
+| customArtifactory.authentication: basic.username: basic.password: apiKey.headerName: apiKey.value: | Used for downloading the JDBC driver using a custom artifactory which is secured with Basic or APIKey Authentication. Use `customArtifactory.authentication.basic` section to provide access credentials or use `customArtifactory.authentication.apiKey` section to provide APIKey value and dedicated APIKey header details. | <ul><li>basic.username: "\<Custom artifactory basic Authentication username\>"</li><li>basic.password: "\<Custom artifactory basic Authentication password\>"</li><li>apiKey.headerName: "\<Custom artifactory dedicated APIKey Authentication header name\>"</li><li>apiKey.value: "\<Custom artifactory APIKey value for APIKey authentication\>"</li> </ul> |
```
Comment (taz-pega-work) [inline docs/Deploying-Pega-on-AKS.md:485]: | customArtifactory.authentication: basic.username: basic.password: apiKey.headerName: apiKey.value: | To download a JDBC driver from your custom artifactory which is secured with Basic or APIKey Authentication. Use `customArtifactory.authentication.basic` section to provide access credentials or use `customArtifactory.authentication.apiKey` section to provide APIKey value and dedicated APIKey header details. | <ul><li>basic.username: "\<Custom artifactory basic Authentication username\>"</li><li>basic.password: "\<Custom artifactory basic Authentication password\>"</li><li>apiKey.headerName: "\<Custom artifactory dedicated APIKey Authentication header name\>"</li><li>apiKey.value: "\<Custom artifactory APIKey value for APIKey authentication\>"</li> </ul> |
| customArtifactory.certificate: | Custom artifactory SSL certificate verification is enabled by default. If your custom artifactory domain has a self-signed SSL certificate, provide the certificate. You can disable SSL certificate verification by setting `customArtifactory.enableSSLVerification` to `false`; however, this setting establishes an insecure connection. | <ul><li> certificate: "\<custom artifactory SSL certificate to be verified\>"</li></ul> |

File: docs/Deploying-Pega-on-EKS.md [hunk 7]
```diff
@@ -512,6 +512,8 @@ Configure the parameters so the pega.yaml Helm chart matches your deployment res
...
    | jdbc.rulesSchema: jdbc.dataSchema:  | Set the names of both your rules and the data schema to the values that Pega Platform uses for these two schemas.      | <ul><li>rulesSchema: "rules" </li><li>dataSchema: "data"</li></ul>  |
+   | customArtifactory.authentication: basic.username: basic.password: apiKey.headerName: apiKey.value: | Used for downloading the JDBC driver using a custom artifactory which is secured with Basic or APIKey Authentication. Use `customArtifactory.authentication.basic` section to provide access credentials or use `customArtifactory.authentication.apiKey` section to provide APIKey value and dedicated APIKey header details. | <ul><li>basic.username: "\<Custom artifactory basic Authentication username\>"</li><li>basic.password: "\<Custom artifactory basic Authentication password\>"</li><li>apiKey.headerName: "\<Custom artifactory dedicated APIKey Authentication header name\>"</li><li>apiKey.value: "\<Custom artifactory APIKey value for APIKey authentication\>"</li> </ul> |
```
Comment (taz-pega-work) [inline docs/Deploying-Pega-on-EKS.md:515]: same minor edits as noted on docs/Deploying-Pega-on-AKS.md.

File: docs/Deploying-Pega-on-GKE.md [hunk 8]
```diff
@@ -293,6 +293,8 @@ Configure the following parameters so the pega.yaml Helm chart matches your depl
...
 | jdbc.rulesSchema: jdbc.dataSchema:  | Set the names of both your rules and the data schema to the values that Pega Platform uses for these two schemas.      | rulesSchema: "rules" dataSchema: "data" |
+| customArtifactory.authentication: basic.username: basic.password: apiKey.headerName: apiKey.value: | Used for downloading the JDBC driver using a custom artifactory which is secured with Basic or APIKey Authentication. Use `customArtifactory.authentication.basic` section to provide access credentials or use `customArtifactory.authentication.apiKey` section to provide APIKey value and dedicated APIKey header details. | <ul><li>basic.username: "\<Custom artifactory basic Authentication username\>"</li><li>basic.password: "\<Custom artifactory basic Authentication password\>"</li><li>apiKey.headerName: "\<Custom artifactory dedicated APIKey Authentication header name\>"</li><li>apiKey.value: "\<Custom artifactory APIKey value for APIKey authentication\>"</li> </ul> |
```
Comment (taz-pega-work) [inline docs/Deploying-Pega-on-GKE.md:296]: same minor edits as noted on docs/Deploying-Pega-on-AKS.md.

File: docs/Deploying-Pega-on-openshift.md [hunk 9]
```diff
@@ -258,6 +258,8 @@ Configure the following parameters so the pega.yaml Helm chart matches your depl
...
 | jdbc.rulesSchema: jdbc.dataSchema:  | Set the names of both your rules and the data schema to the values that Pega Platform uses for these two schemas.      | rulesSchema: "rules" dataSchema: "data" |
+| customArtifactory.authentication: basic.username: basic.password: apiKey.headerName: apiKey.value: | Used for downloading the JDBC driver using a custom artifactory which is secured with Basic or APIKey Authentication. Use `customArtifactory.authentication.basic` section to provide access credentials or use `customArtifactory.authentication.apiKey` section to provide APIKey value and dedicated APIKey header details. | <ul><li>basic.username: "\<Custom artifactory basic Authentication username\>"</li><li>basic.password: "\<Custom artifactory basic Authentication password\>"</li><li>apiKey.headerName: "\<Custom artifactory dedicated APIKey Authentication header name\>"</li><li>apiKey.value: "\<Custom artifactory APIKey value for APIKey authentication\>"</li> </ul> |
```
Comment (taz-pega-work) [inline docs/Deploying-Pega-on-openshift.md:261]: same minor edits as noted on docs/Deploying-Pega-on-AKS.md.

**Review by taz-pega-work — COMMENTED** [review]
made editorial comments after discussing some details with Heshma.

**Review by taz-pega-work — APPROVED** [review]
great adaptation! looks perfect. 

Thanks for all of your attention here!!

**Review by pega-talba — APPROVED** [review]

**Review by misterdorito — APPROVED** [review]

**Review by APegaDavis — APPROVED** [review]

**Pull request merged**